src/contracts/AbstractARM.sol

```diff
@@ -726,4 +772,136 @@ abstract contract AbstractARM is OwnableOperable, ERC20Upgradeable {
 
         fees = SafeCast.toUint256(assetIncrease) * fee / FEE_SCALE;
     }
+
+    ////////////////////////////////////////////////////
+    ///         Lending Market Functions
+    ////////////////////////////////////////////////////
+
+    /// @notice Owner adds supported lending market to the ARM.
+    /// In order to be a safe lending market for the ARM, it must be:
+    ///  1. up only exchange rate
+    ///  2. no slippage
+    ///  3. no fees.
+    /// @param _markets The addresses of the lending markets to add
+    function addMarkets(address[] calldata _markets) external onlyOwner {
+        for (uint256 i = 0; i < _markets.length; i++) {
+            address market = _markets[i];
+            require(market != address(0), "ARM: invalid market");
+            require(!supportedMarkets[market], "ARM: market already supported");
+            require(IERC4626(market).asset() == liquidityAsset, "ARM: invalid market asset");
+
+            supportedMarkets[market] = true;
+
+            emit MarketAdded(market);
+        }
+    }
+
+    /// @notice Owner removes a supported lending market from the ARM.
+    /// This can not be the active market.
+    /// @param _market The address of the lending market to remove
+    function removeMarket(address _market) external onlyOwner {
+        require(_market != address(0), "ARM: invalid market");
+        require(supportedMarkets[_market], "ARM: market not supported");
+        require(_market != activeMarket, "ARM: market in active");
+
+        supportedMarkets[_market] = false;
+
+        emit MarketRemoved(_market);
+    }
+
+    /// @notice set a new active lending market for the ARM.
+    /// This can be set to address(0) to disable the use of a lending market.
+    function setActiveMarket(address _market) external onlyOperatorOrOwner {
+        require(_market == address(0) || supportedMarkets[_market], "ARM: market not supported");
+        // Read once from storage to save gas and make it clear this is the previous active market
+        address previousActiveMarket = activeMarket;
+        require(previousActiveMarket != _market, "ARM: already active market");
+
+        if (previousActiveMarket != address(0)) {
+            // Redeem all shares from the previous active lending market.
+            // balanceOf is used instead of maxRedeem to ensure all shares are redeemed.
+            // maxRedeem can return a smaller amount of shares than balanceOf if the market is highly utilized.
+            uint256 shares = IERC4626(previousActiveMarket).balanceOf(address(this));
+            // This could fail if the market has high utilization
+            if (shares > minSharesToRedeem) {
```

Comment:
This if statement breaks our up only invariant and the defense beyond that to a loss of user funds has to do with gas prices and amount of funds in our contract. While we're probably good there, it's not something I like very much.

What's the harm in removing this if in `setActiveMarket`? Worse case is that we have to send a tiny amount of shares to unlock a switch from an almost empty lending market? This could be done by the operator, or by allocating more in.

(The minSharesToRedeem if statement in `_allocate` is fine, since it doesn't affect accounting if it runs or not)

Reply:
It would still be fine to have an if shares greater than zero check here.